### ISSUE TITLE
SC users is not clear, that e.g. shops are open by default on public holidays, when they add opening_hours=*

### ISSUE THREAD

**Issue by fabi003** [streetcomplete/StreetComplete#5455]
[Here](https://www.openstreetmap.org/node/3644399093/history) e.g. an SC 54.0 user added the opening_hours=* and I had to fix the shop again, as it is closed on public holidays, as most of the shops in Germany.

SC should have an "maybe open on PH"-list by country/region, as this, but at least in Germany, this is a complex issue and e.g in Berlin exists exceptions, if an shop, which is normal closed on PH, is located e.g. in a train station. So these lists should not enforced to the users.

So my idea to solve this issue (I have not seen the SC opening_hours editor) is to add an checkbox with "closed on public holidays" which should be pre-checked most times and if the user unchecked these checkbox, they should enter the opening_hours=* on PH and if possible the times should be transformed into a correct rule, as e.g. "; PH off", which is in most cases a good default rule, overwrites times if rules with opening hours after midnight are used and the PH closing rule must then written as a rule for the longest opening time range e.g. "Mo-Th 18:00-03:00, Fr-Sa 20:00-06:00; PH 18:00-06:00 off".

Preliminary "maybe open on PH"-list for Germany (in general);
shop=newsagent, florist, bakery, pastry, greengrocer, dairy
amenity=fuel, pharmacy, fast_food, restaurant, cafe, ice_cream, bar, pub, biergarten, nightclub

...and shops on airports (aeroway=terminal and aeroway=aerodrome), habours of ferries (amenity=ferry_terminal) and train stations.

**Comment by Helium314** [comment]
Users will most likely enter what they see signed on site. If there is no specific mention of public holidays, I would not expect anyone to explicitly specify it as open or closed (both is already possible).

see also https://github.com/streetcomplete/StreetComplete/issues/2332#issuecomment-736513222

**Comment by matkoniecz** [comment]
Is it typical in Germany to have clear statement whether given POI is open on PH, as part of regular opening hours?

**Comment by HolgerJeromin** [comment]
> Is it typical in Germany to have clear statement whether given POI is open on PH, as part of regular opening hours?

No. I would say less than 10% mention PH. 
But most things (99% of restaurants, bakeries and other shops) will copy the sunday hours.

**Comment by westnordost** [comment]
As with everything in SC, mappers are expected to map what they see. In this case, write the opening hours as they see them on the sign. If anything is not specifically mentioned, it is not added. E.g. some restaurants will write "Mo off" explicitly, others write "Tu-Fr ...".

See https://github.com/streetcomplete/StreetComplete/issues/1825#issuecomment-627939340 for a longer explanation for the reason why I actively advise against specifying the `PH` if it is not explicitly written on the sign.

**Comment by fabi003** [comment]
@westnordost
You won the closing-before-thinking-award again! As no one seems to look at the posted examples of SC modifications in the first post in this ticket and in #5454. 

How should the SC mapper know, that he has to specify the, in most times required, default public holiday closing which is OSM specific, but not to the general public,, when he enters the opening hours from the sign which in most times contain no times for PH opening?

**Comment by matkoniecz** [comment]
> You won the closing-before-thinking-award again!

I highly encourage you to avoid insults. This is unlikely to work even with people obligated to follow your orders or people paid by you.

It is extremely unlikely to work in cases where you try to persuade someone to change their mind. Or when you asks someone to do something for you.

(I stopped reading, wrote this comment and will silence this issue without reading further)

**Comment by fabi003** [comment]
StreetComplete developer questioner:
 
☐  I have really tried to understand the reported issue from the example in the first post.
☐  I have asked, to further explain the reported issue, if the report was not clear enough.
☐  I have not closed the issue, due it is a legitimate problem of the StreetComplete app.
☐  I have not closed other legitimate issues reports for the SC app in the past.
☒  I maybe develop SC in my free time or unpaid and have enough other things to do, but as some closed issue reports are valid, I should leave them open for someone to help me to fix them.

**Comment by mnalis** [comment]
> You won the closing-before-thinking-award again!

@fabi003 Please don't be rude. That can accomplish no good for anyone, and only ever portrays the person being rude in bad light - no matter what the argument might have been. There is good essay on [Wikipedia:No angry mastodons](https://en.wikipedia.org/wiki/WP:NAM) about how to handle such moments when we get angry. I myself have found it useful in the past.

Also, just because the issue is closed, does not mean the discussion cannot continue, and sometimes issues get re-opened if further discussion finds reasons to do so. There are reasons why some issues get closed early. You may not know them, or you may disagree with them, but if you expect to get any help from other people in community, you should _at the very least_ remain courteous to them in all times, even if (and _especially_ if) you disagree with them or get frustrated.

> As no one seems to look at the posted examples of SC modifications in the first post in this ticket and in https://github.com/streetcomplete/StreetComplete/issues/5454.

***I*** at least definitely _did_ look in posted examples and #5454 and even asked for clarifications (some of which you have not replied yet), so you are being rude _and_ untruthful there. But I'll give you a second chance to get your act back together - but please forego the rudeness and accusations.  So, to get back to issue at hand:

> So my idea to solve this issue (I have not seen the SC opening_hours editor) 

I would suggest you do look at it. It is almost impossible to make a reasonable suggestion on how to improve something, if you **haven't even looked** how that something currently looks.

> So my idea to solve this issue (I have not seen the SC opening_hours editor) is to add an checkbox with "closed on public holidays" which should be pre-checked most times and if the user unchecked these checkbox, they should enter the opening_hours=* on PH and if possible the times should be transformed into a correct rule, as e.g. "; PH off",

There is indeed already an option to `Add off days`, where you can choose `PH` among other things, and which would result in StreetComplete appending `; PH off` to opening hours. It is not marked by "default" for the same reason there is no "default" tagging of `Mo-Fr 09:00-17:00` or any other popular working hours -- because the mapper is expected to **explicitly** transcribe **any and all** signed `opening_hours` one by one, and we want to avoid errors-by-forgetting-to-remove-default. 

StreetComplete mapper is not expected to even know (and much less tag) what is "usual" or "common", but **only things that are explicitly signed** (because many mappers are not actually residents of that country, but tourists, who have no idea about local customs). That follows ["Map what's on the ground"](https://wiki.openstreetmap.org/wiki/Good_practice#Map_what's_on_the_ground) OSM best practice.

> which is in most cases a good default rule, overwrites times if rules with opening hours after midnight are used and the PH closing rule must then written as a rule for the longest opening time range e.g. "Mo-Th 18:00-03:00, Fr-Sa 20:00-06:00; PH 18:00-06:00 off".

I'm not quite sure what you've meant here. That specific opening_hours example that you mention here would never be tagged by StreetComplete as far as I know. In cases where StreetComplete `opening_hours` editor is not good enough to specify exactly the raw `opening_hours` tag that user want; it is expected that user will leave a public OSM Note instead (preferably with a picture). That option can be found under `Other answers` / `Can't say` menu.

> Preliminary "maybe open on PH"-list for Germany (in general);
shop=newsagent, florist, bakery, pastry, greengrocer, dairy
amenity=fuel, pharmacy, fast_food, restaurant, cafe, ice_cream, bar, pub, biergarten, nightclub

I'm not very sure what you've meant here either. Are you saying that you think that some new (tag or tag value) should be invented for _"maybe open on PH"_? Note that unless syntax is currently clearly defined on [opening_hours wiki](https://wiki.openstreetmap.org/wiki/Key:opening_hours), **nothing can be done about it in that idea in this StreetComplete issue tracker**, but you need to suggest and discuss it on [Tagging ML](https://wiki.openstreetmap.org/wiki/Tagging_mailing_list) or [Community tagging category](https://community.openstreetmap.org/c/general/tagging/70)  first. Only where there is global community consensus on that tagging, may it be suggested again in StreetComplete issue tracker (preferably in a new issue).

---

In short - If those signs however do follow `opening_hours` wiki, are **popular enough**, and you have suggestion how to tag them, then (to improve understanding) could you please post **pictures of those signs**, and accompany each picture with `opening_hours=xxxxx` that you think should accompany it? 

That should be best way to get message across without any misunderstandings and frustrations. Thank you. (that suggestion applies to anybody else who may have similar issues too)

**Comment by fabi003** [comment]
Thanks that after the reports in #1825 and #276 this problem was fixed with #5463. Don't know that is was work in progress (#5411).
 Looks to my as if RFC 9225 applies here.

**Comment by westnordost** [comment]
No, it was fixed looong before. #5463 has nothing to do with this.
